blockdag/error.go

```diff
@@ -121,6 +121,11 @@ const (
 	// either does not exist or has already been spent.
 	ErrMissingTxOut
 
+	// ErrDoubleSpendsWithBlockTransaction indicates a transaction
+	// that spends an output that was already spent by another
+	// transaction in the same block.
+	ErrDoubleSpendsWithBlockTransaction
```

Comment:
I don't like this name too much, is a bit confusing.
How about ErrDoubleSpendWithinBlock? Or ErrDoubleSpendInSameBlock or something else that makes it clear what this is?